.github/workflows/aws_nightly_cleanup.yml

```diff
@@ -17,29 +21,83 @@ on:
 
 env:
     AWS_PROFILE: infex
-    CLEANUP_OLDER_THAN: ${{ github.event.inputs.cleanup_older_than || '12h' }}
     # renovate: datasource=github-tags depName=gruntwork-io/cloud-nuke
     CLOUD_NUKE_VERSION: v0.37.2
 
-# limit to a single execution per ref of this workflow
+# Limit workflow to a single execution per ref
 concurrency:
     group: ${{ github.workflow }}-${{ github.ref }}
     cancel-in-progress: true
 
 jobs:
-    aws-nightly-cleanup:
+    aws-cleanup:
         runs-on: ubuntu-latest
         strategy:
             fail-fast: false # don't propagate failing jobs
             matrix:
-                aws_test_regions: [eu-west-2, eu-west-3]
-        env:
-            AWS_REGION: ${{ matrix.aws_test_regions }}
+                # Define regions and types of cleanup based on day
+                # Please also update the README.md of the project
+                config:
+                    - region: eu-west-2
+                      day: All
+                      cleanup_older_than: 12h
+                    - region: eu-west-3
+                      day: All
+                      cleanup_older_than: 12h
+                    - region: eu-north-1
+                      day: Saturday
+                      cleanup_older_than: 0h
+                    - region: us-east-1
+                      day: Saturday
+                      cleanup_older_than: 0h
+
         steps:
             - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
 
+            - name: Determine AWS_REGION and CLEANUP_OLDER_THAN
+              id: determine-values
+              run: |
+                  if [[ "${{ github.event_name }}" == "workflow_dispatch" ]]; then
+                      AWS_REGION="${{ github.event.inputs.region }}"
+                      CLEANUP_OLDER_THAN="${{ github.event.inputs.cleanup_older_than }}"
+                  else
+                      AWS_REGION="${{ matrix.config.region }}"
+                      CLEANUP_OLDER_THAN="${{ matrix.config.cleanup_older_than }}"
+                  fi
+
+                  echo "AWS_REGION=$AWS_REGION" | tee -a "$GITHUB_ENV"
+                  echo "CLEANUP_OLDER_THAN=$CLEANUP_OLDER_THAN" | tee -a "$GITHUB_ENV"
+
+            - name: Check if job should run based on the day
+              id: day-check
+              run: |
+                  # Initialize a variable to determine if the job should continue
+                  should_run=true
+
+                  # If it's a workflow_dispatch, only allow the first region (eu-west-2) of the matrix to run
+                  if [[ "${{ github.event_name }}" == "workflow_dispatch" ]]; then
+                      if [[ "${{ matrix.config.region }}" != "eu-west-2" ]]; then
+                          echo "Skipping job for this region because only the first region is allowed to run for workflow_dispatch."
+                          should_run=false
```

Comment:
doesn't that mean the input of the workflow dispatch is useless and could be removed?

Reply:
I'm not sure we're on the same page.

Workflow_dispatch:
input: eu-west-2 --> runs cleanups of that particular region
input: eu-west-3 --> exit 1
input: eu-north-1 --> exit 1
input: us-west-1 --> exit 1

I can not use it to clean up any of the other regions. If it's intended, then my question remains: why have the input for the workflow dispatch if we only limit it to the eu-west-2 region anyway?

---

Just an idea, doesn't have to be implemented:
Wouldn't it have been easier to create a job prior that creates a dynamic matrix based on `day-check` / workflow_dispatch.
For the weekly cleanup, could just be a condition that `region == north && day == saturday` as otherwise day is undefined or not the same value.

Reply:
wouldn't it have to be?
```
if [[ "${{ matrix.config.region }}" != "$AWS_REGION" ]];
```

Reply:
ahhhh, okay the region running has nothing to do with the region executing.
The label is slightly misleading.
Got it now, all good!